### ISSUE TITLE
[SECURITY] Releases are built/executed/released in the context of insecure/untrusted code

### ISSUE THREAD

**Issue by JLLeitschuh** [HellFirePvP/AstralSorcery#1009]
[CWE-829: Inclusion of Functionality from Untrusted Control Sphere](https://cwe.mitre.org/data/definitions/829.html)

All of these build files include resolving dependencies over HTTP instead of HTTPS. Any of these artifacts could have been MITM to maliciously compromise them and infect the build artifacts that were produced. Additionally, if any of these JAR files were compromised, any developers using these could continue to be infected past updating to fix this.

https://github.com/HellFirePvP/AstralSorcery/blob/233b529274d4be9c0a88913233a5492909f66887/build.gradle#L4-L7

https://github.com/HellFirePvP/AstralSorcery/blob/233b529274d4be9c0a88913233a5492909f66887/build.gradle#L41-L70

This vulnerability has a CVSS v3.0 Base Score of 8.1/10
https://nvd.nist.gov/vuln-metrics/cvss/v3-calculator?vector=AV:N/AC:H/PR:N/UI:N/S:U/C:H/I:H/A:H

This isn't just theoretical; POC code exists already to maliciously compromise jar file inflight.

See:
* https://max.computer/blog/how-to-take-over-the-computer-of-any-java-or-clojure-or-scala-developer/
* https://github.com/mveytsman/dilettante

To fix: Update your `build.gradle` so that artifacts are resolved over HTTPS instead of HTTP.

**Comment by HellFirePvP** [comment]
"on hold" more or less until JEI's maven repo either allows for a secure connection or is hosted somewhere else that allows for that..

**Comment by HellFirePvP** [comment]
There's seemingly at least some movement to move to HTTPS on the last maven repository i'm waiting on, however this is with that kinda resolved. Will keep an eye on it and switch as soon as it's available through more secure means. Sharing the same opinion as KingLemming from the issue posted on CoFH/Feedback about the same thing.
Closing this as i don't deem it necessary to keep it open.
